impeller/renderer/backend/vulkan/allocator_vk_unittests.cc

```diff
@@ -71,6 +73,19 @@ TEST(AllocatorVKTest, MemoryTypeSelectionTwoHeap) {
   EXPECT_EQ(AllocatorVK::FindMemoryTypeIndex(4, properties), -1);
 }
 
+TEST(AllocatorVKTest, DeviceBufferCoherency) {
+  auto const context = MockVulkanContextBuilder().Build();
+  auto allocator = context->GetResourceAllocator();
+
+  std::shared_ptr<DeviceBuffer> buffer =
+      allocator->CreateBuffer(DeviceBufferDescriptor{
+          .storage_mode = StorageMode::kHostVisible,
+          .size = 1024,
+      });
+
+  EXPECT_TRUE(DeviceBufferVK::Cast(*buffer).IsHostCoherent());
```

Comment:
do you have a similar test for when the flag isn't `type.propertyFlags & vk::MemoryPropertyFlagBits::eHostCoherent`?

Reply:
So this test is using a mocked out vulkan instance which is ... kind of difficult to configure. I could easily add a real test that uses swiftshader to allocate buffers, but those will always be non coherent.

Reply:
Done

Reply:
well shoot, CI only has non-coherent memory.

Reply:
🤷‍♂️